src/Codeception/Lib/Driver/MongoDb.php

```diff
@@ -1,83 +1,105 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Codeception\Lib\Driver;
 
 use Codeception\Exception\ModuleConfigException;
 use Codeception\Exception\ModuleException;
+use Exception;
+use MongoClient;
+use MongoConnectionException;
 use MongoDB\Database;
+use MongoException;
 
 class MongoDb
 {
+    /**
+     * @var int
+     */
     const DEFAULT_PORT = 27017;
-
-    private $legacy;
+    /**
+     * @var bool
+     */
+    private $isLegacy;
+    /**
+     * @var \Codeception\Lib\Driver\MongoDB|null
+     */
     private $dbh;
-    private $dsn;
+    /**
+     * @var string|null
+     */
     private $dbName;
     private $host;
     private $user;
     private $password;
+    /**
+     * @var \MongoDB\Client|MongoClient|null
+     */
     private $client;
+    /**
+     * @var string
+     */
     private $quiet = '';
 
-    public static function connect($dsn, $user, $password)
+    public static function connect($dsn, $user, $password): void
     {
-        throw new \Exception(__CLASS__ . '::connect() - hm, it looked like this method had become obsolete...');
+        throw new Exception(__CLASS__ . '::connect() - hm, it looked like this method had become obsolete...');
     }
 
     /**
      * Connect to the Mongo server using the MongoDB extension.
      */
-    protected function setupMongoDB($dsn, $options)
+    protected function setupMongoDB($dsn, $options): void
     {
         try {
             $this->client = new \MongoDB\Client($dsn, $options);
             $this->dbh    = $this->client->selectDatabase($this->dbName);
-        } catch (\MongoDB\Driver\Exception $e) {
-            throw new ModuleException($this, sprintf('Failed to open Mongo connection: %s', $e->getMessage()));
+        } catch (\MongoDB\Driver\Exception $exception) {
+            throw new ModuleException($this, sprintf('Failed to open Mongo connection: %s', $exception->getMessage()));
         }
     }
 
     /**
      * Connect to the Mongo server using the legacy mongo extension.
      */
-    protected function setupMongo($dsn, $options)
+    protected function setupMongo($dsn, $options): void
     {
         try {
```

Comment:
I think that legacy extension is incompatible with PHP 7: https://www.php.net/manual/en/mongo.installation.php
This code can be deleted.